sql/core.go

```diff
@@ -341,6 +341,18 @@ type CheckAlterableTable interface {
 	DropCheck(ctx *Context, chName string) error
 }
 
+// StatementBounder allows the boundaries of an INSERT, REPLACE, UPDATE, or DELETE statement to be passed along.
+// This is modeled after the SAVEPOINT statements, but are local and specific to the aforementioned data manipulation
+// statement boundaries.
+type StatementBounder interface {
```

Comment:
I don't like the overloading with transaction semantics here

How about instead:

Savepoint => StatementBegin
Rollback => ErrorEncountered / DiscardChanges?
Release => StatementComplete